src/transformers/utils/quantizers.py

```diff
@@ -0,0 +1,869 @@
+
+        elif self.quantization_method == QuantizationMethod.GPTQ:
+            return GPTQHFQuantizer
+
+        elif self.quantization_method == QuantizationMethod.AWQ:
+            return AWQHFQuantizer
+
+    def get_quantizer(self, config: PretrainedConfig) -> HFQuantizer:
+        """
+        Process model config if any.
+        # Assumes that config.quantization_config is None or subclass of QuantizationConfigMixin
+
+        Returns:
+            Quantizer instance to use for model loading.
+        """
+        if not hasattr(self, "quantization_method"):
+            raise AttributeError("quantization_config from args must be parsed before parsing model config")
+
+        if hasattr(config, "quantization_config"):
+            # the model is considered PREQUANTIZED and config from the model prevails
+            config_quantization_config = self.validate_quantization_config(config.quantization_config)
+            quantization_method_from_config = getattr(config_quantization_config, "quant_method")
+
+            if self.quantization_config is not None:
+                self.quantization_config = self.handle_configs_collision(config_quantization_config)
+            else:
+                self.quantization_config = config_quantization_config
+
+            self.quantization_method = quantization_method_from_config
+            self.quantization_status = QuantizationStatus.PREQUANTIZED
+
+        elif self.quantization_method == QuantizationMethod.AWQ:
+            raise ValueError(
+                "You cannot quantize with AWQ a non-quantized model using transformers, please refer to the quantization documentation"
+                " to read more about how to quantize models with AWQ algorithm https://huggingface.co/docs/transformers/main_classes/quantization"
+            )
+
+        elif self.quantization_config is not None:
+            # using quantization_config from args for fresh quantization
+            self.quantization_status = QuantizationStatus.FRESH
+
+        else:
+            # Normal loading without quantization. No quant arguments present in args or model config.
+            return
+
+        config.quantization_config = self.quantization_config
+        quantizer_class = self.get_quantizer_class()
+        quantizer = quantizer_class(
+            quantization_config=self.quantization_config,
+            quantization_status=self.quantization_status,
+        )
+        return quantizer
+
+
+def get_module_from_name(module, tensor_name: str) -> Tuple[Any, str]:
+    if "." in tensor_name:
+        splits = tensor_name.split(".")
+        for split in splits[:-1]:
+            new_module = getattr(module, split)
+            if new_module is None:
+                raise ValueError(f"{module} has no attribute {split}.")
+            module = new_module
+        tensor_name = splits[-1]
+    return module, tensor_name
+
+
+class HFQuantizer(ABC):
```

Comment:
If yes, then we should make sure it follows our usual logic: 
1. A quantizer is kind of like a model. We need a better name than HFQuantizer. 
2. We could just load a HFQuantiizer from pretrained, kind of what we do for a model we get the config and init it, we get the quantization method and init whatever elle you need. 
3.

Reply:
I viewed HFQuantizer as a tool, and not a model. Something like a `peft` thing.  It may exist separately from models and potentially get passed as an argument in `.from_pretrained()`. So a user would take a custom quantizer (subclassed from HFQuantizer) and load a model using it.

I even thought about making a simple example of like "RoundToNearestQuantizer", but my CUDA skills are not strong yet.

Name-wise - I had no other good ideas. What would you suggest? `QuantizationWrapper`?